simulation/behavior_tree_plugin/src/vehicle/follow_trajectory_sequence/follow_polyline_trajectory_action.cpp

```diff
@@ -56,16 +84,22 @@ auto FollowPolylineTrajectoryAction::providedPorts() -> BT::PortsList
 
 auto FollowPolylineTrajectoryAction::tick() -> BT::NodeStatus
 {
+  if (!trajectory_follower)
+  {
+    trajectory_follower = std::make_unique<PositionModePolylineTrajectoryFollower>();
+  }
+
   if (getBlackBoardValues();
       request != traffic_simulator::behavior::Request::FOLLOW_POLYLINE_TRAJECTORY or
-      not getInput<decltype(trajectory_parameter)>(
-        "polyline_trajectory_parameter", trajectory_parameter) or
+      not getInput<decltype(trajectory_parameter)>("polyline_trajectory_parameter", trajectory_parameter) or
       not getInput<decltype(target_speed)>("target_speed", target_speed) or
       not trajectory_parameter) {
     return BT::NodeStatus::FAILURE;
   } else if (
+
     const auto updated_status =
-      makeUpdatedStatus(entity_status, trajectory_parameter, behavior_parameter, step_time)) {
+      trajectory_follower->followTrajectory(entity_status, trajectory_parameter, behavior_parameter, step_time)) {
```

Comment:
po calym review: z tego co widze to wszystkie te parametry aktualizowane sa przez getBlackBoardValues(), wiec faktycznie trzeba je przekazywac.. (bo moga sie zmienic) - wiec wycofuje zwiazane z tym komentarze

Reply:
tylko ze idac tym tropem nie wiem czy jest sens tworzyc trajectory_follower wczesniej, zamiast po prostu tutaj przed i byloby tak:
```suggestion
   trajectory_follower = std::make_unique<PositionModePolylineTrajectoryFollower>(entity_status, behavior_parameter, step_time);
      trajectory_follower->followTrajectory(trajectory_parameter) 
```
przemysl to prosze